CHANGELOG.md

```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+## 4.0.0
+
+This is a complete rewrite of the crate. It focuses on the `Mapping` type, which
+provides high level metadata about a proguard mapping file, and allows iterating
+over the contained `Record`s, and the `Mapper`, which allows re-mapping class
+names and entire frames, with support for inlined frames as well.
```

Comment:
How would you feel about giving a few examples on how to migrate (e.g. which type to replace with which)?